src/java/org/apache/nutch/fetcher/FetcherReducer.java

```diff
@@ -861,8 +861,8 @@ public void run(Context context) throws IOException, InterruptedException {
       LOG.info("Fetcher: throughput threshold sequence: "
           + throughputThresholdSequence);
     }
-    long throughputThresholdTimeLimit = conf.getLong(
-        "fetcher.throughput.threshold.check.after", -1);
```

Comment:
This would set the time when the throughput threshold is check per task not per job. It makes a difference if a task is started later or is restarted. In analogy to "fetcher.timelimit.mins" and how it's done in 1.x, I would set it (overwrite it with the effective system time) in `FetcherJob.run(Map<String, Object> args)`.

Reply:
Sorry, after thinking again about it - you're right. Fetcher differs between 1.x and 2.x in one important point:
- (1.x) fetching is taking place in the map phase
- (2.x) pages are fetched in the reduce phase.
Ok, the property `fetcher.timelimit.mins` is set absolutely for the whole job. For scheduling cycles it's important that the end time is per job and not per task. For 2.x users must include the map phase. The throughput threshold check is not really about limiting the time spent for one crawl/fetch cycle but to avoid tasks consuming resources if there are only few or slow queues. It's definitely ok to set this limit relative per reduce task.